docs/proposals-accepted/202207-granular-endpoint-configuration

```diff
@@ -0,0 +1,58 @@
+# Granular Endpoint Configuration
+
+* **Related Tickets:**
+
+  * [Per-Store TLS configuration in Thanos Query](https://github.com/thanos-io/thanos/issues/977)
+
+## What
+
+This proposal builds up on the [Unified Endpoint Discovery Proposal](https://thanos.io/tip/proposals-accepted/202101-endpoint-discovery.md/) and [Automated, per-endpoint mTLS Proposal](https://thanos.io/tip/proposals-accepted/202106-automated-per-endpoint-mtls.md/)
+
+## Why
+
+The Thanos Querier component supports basic mTLS configuration for internal gRPC communication. Mutual TLS (mTLS) ensures that traffic is both secure and trusted in both directions between a client and server. This works great for basic use cases but it still requires extra forward proxies to make it work for bigger deployments.
+
+Let’s imagine we have an Observer Cluster that is hosting Thanos Querier along with Thanos Store Gateway. Within the same Observer Cluster, we would like to connect one or more Thanos Sidecars. Additionally, we also want to connect the Querier in the Observer Cluster to several remote instances of Thanos Sidecar in remote clusters.
+
+In this scenario we need to use a proxy server (e.g. envoy). But, it would probably be simpler to do away with the proxies and let Thanos speak directly and securely with the remote endpoints.
+
+## Pitfalls of the current solution
+
+Ideally, we would want to use mTLS to encrypt the connection to the remote clusters. If we would enable the current mTLS, it would be applied to all the storeAPI’s but we don’t want it to be applied on the storeAPI’s of central Thanos instance (Observer cluster) in which Thanos query component is present (for faster communications with storeAPI’s (sidecars) of same cluster, to reduce the pain of provisioning certificates etc.). So it requires extra forward proxies to make it work for bigger (multi-cluster) deployments.
+
+## Goals
+
+* To add support for per-endpoint TLS configuration in Thanos Query Component for internal gRPC communication
+
+## Non-Goals
+
+## How
+
+A new CLI option `--endpoints.config`, with no dynamic reloading, which will accept the path to a yaml file is proposed which contains a list as follows :
+
+```go
+- tls_config:
+    cert_file: ""
+    key_file: ""
+    ca_file: ""
+    server_name: ""
+  endpoints: []
+  endpoints_sd_files:
+    - files: []
+  mode: ""
+```
```

Comment:
Let's maybe add a field for filtering APIs as mentioned in https://github.com/thanos-io/thanos/pull/4785/files#diff-5d0883e9f6db360a29dd686b1ba6dbf152d0ab9549a0506aeffdf532f5ae0591R23. Something where you can specify the APIs Querier needs to access, and explain that below! 

For implementing this proposal, maybe we can keep filtering as a TODO and focus on per-endpoint config and flag deprecation instead. We can then implement filtering later on! WDYT?